module/zfs/spa.c

```diff
@@ -2679,6 +2680,19 @@ spa_load_impl(spa_t *spa, uint64_t pool_guid, nvlist_t *config,
 		return (spa_load(spa, state, SPA_IMPORT_EXISTING, B_TRUE));
 	}
 
+	/* Grab the secret checksum salt from the MOS. */
+	error = zap_lookup(spa->spa_meta_objset, DMU_POOL_DIRECTORY_OBJECT,
```

Comment:
This isn't really secret. Any user with sudo can just look this up with zdb, in fact. I would not be comfortable using this as a MAC while the 'key' is so easy to figure out.